admin/collection_providers/views.py

```diff
@@ -233,36 +249,19 @@ class CollectionProviderChangeForm(PermissionRequiredMixin, UpdateView):
     model = CollectionProvider
     form_class = CollectionProviderForm
 
-    def form_valid(self, form):
-        if self.object.primary_collection:
-            self.object.primary_collection.collected_type_choices.extend(form.cleaned_data['collected_type_choices']['added'])
-            for item in form.cleaned_data['collected_type_choices']['removed']:
-                self.object.primary_collection.collected_type_choices.remove(item)
+    def process_choices(self, choices_name, form):
+        collection = self.object.primary_collection
+        choices_added = form.cleaned_data[f'{choices_name}_choices']['added']
```

Comment:
This feels like a mildly over-DRY approach. Let's just pass the full field name so that we don't have to construct it manually in all of these places.